index.js

```diff
@@ -47,23 +48,40 @@ class CircuitBreaker {
      * Retry wrapper for the circuit breaker to retry on failure as long as circuit is closed
      * @method runCommand
      * @param  {arguments}   arguments           List of arguments to call the circuit breaker command with
-     * @param  {Function}    callback            Last argument is the callback to callback upon completion
+     * @param  {Function}    [callback]          Last argument is the callback to callback upon completion
      */
     runCommand() {
         const args = Array.prototype.slice.call(arguments);
-        const callback = args.pop();
+        let callback;
+
+        if (typeof args[args.length - 1] === 'function') {
+            callback = args.pop();
+        }
+
         const wrapBreaker = (cb) => {
             this.breaker.apply(this.breaker, args)
-            .then((data) => cb(null, data), (err) => cb(err));
+            .then(data => cb(null, data), err => cb(err));
         };
         const shouldRetry = this.shouldRetry;
 
-        retryFn({
-            method: wrapBreaker,
-            context: this,
-            options: this.retryOptions,
-            shouldRetry: (err) => err && this.isClosed() && shouldRetry(err, args)
-        }, callback);
+        return new Promise((resolve, reject) => {
+            retryFn({
+                method: wrapBreaker,
+                context: this,
+                options: this.retryOptions,
+                shouldRetry: err => err && this.isClosed() && shouldRetry(err, args)
+            }, (err, ...data) => {
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_operator

Converts all arguments after the first into an array.